pkg/pillar/cmd/zedrouter/server.go

```diff
@@ -336,3 +345,51 @@ func (hdl openstackHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	w.WriteHeader(http.StatusNotFound)
 }
+
+// ServeHTTP for kubeConfigHandler provides cluster kube config
+func (hdl kubeConfigHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		msg := "kube config handler: request method is not Post"
+		log.Error(msg)
+		http.Error(w, msg, http.StatusMethodNotAllowed)
+		return
+	}
+	if r.Header.Get("Content-Type") != "application/json" {
+		msg := "kube config handler: Content-Type header is not application/json"
+		log.Error(msg)
+		http.Error(w, msg, http.StatusUnsupportedMediaType)
+		return
+	}
+	// Use http.MaxBytesReader to enforce a maximum read of 32KB from the
+	// response body. A request body larger than that will now result in
+	// Decode() returning a "http: request body too large" error.
+	r.Body = http.MaxBytesReader(w, r.Body, 32768)
+	dec := json.NewDecoder(r.Body)
+	var kubeConfig types.KubeConfig
+	err := dec.Decode(&kubeConfig)
+	if err != nil {
+		switch {
+		case err.Error() == "http: request body too large":
+			msg := "kube config handler: request body must not be larger than 1MB"
+			log.Error(msg)
+			http.Error(w, msg, http.StatusRequestEntityTooLarge)
+		default:
+			msg := fmt.Sprintf("kube config handler: decoding failed: %v", err)
+			log.Errorf(msg)
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		}
+		return
+	}
+	var appInstMetaData types.AppInstMetaData
+	appInstMetaData.AppInstUUID, _ = uuid.FromString(kubeConfig.AppInstUUID)
```

Comment:
I made a comment over a week ago (don't know what happened to that comment) that it doesn't make any sense for the application to send the UUID. Insterad zedrouter/EVE should add the UUID. (Note that the other services use the source IP address to determine the appinstance using lookupAppNetworkStatusByAppIP.)

Reply:
That changes are not committed yet. Waiting for the approval of the EVE community. Sent out proposal yesterday.